### PR TITLE
package_manager: Added pattern for remove packages

### PR DESCRIPTION
**Pull request by petr-balogh**
* Updated remove method which now can takes pattern for removing more
  packages.
* Added list method which list all isntalled packages on host
* Added tests for package_manager

Change-Id: I0d381db9b1bf7b3aec9b250e438233b2ee1f0d87

### CODE REVIEW

File: rrmngmnt/package_manager.py [hunk 1]
```diff
@@ -62,6 +63,31 @@ def exist(self, package):
...
+            "Getting all instaled packages from host %s", self.host
+        )
+        rc, out, err = self.host.executor().run_cmd(cmd)
+        if not rc:
+            return out.split('\n')
+        self.logger.error(
+            "Failed to get installed packages on host %s, rc: %s, err: %s",
+            self.host, rc, err
+        )
+        return []
```
Comment (lukas-bednar) [inline rrmngmnt/package_manager.py]: I prefer to raise exception with all info inside  ... how the user's program using this library is able to figure difference between error and no packages?

File: rrmngmnt/package_manager.py [hunk 2]
```diff
@@ -62,6 +63,31 @@ def exist(self, package):
...
+        :return: installed packages
+        :rtype: list
+        :raise: NotImplementedError
+        """
+        if not self.list_command_d:
+            raise NotImplementedError(
+                "There is no 'list_' command defined."
+            )
+        cmd = self.list_command_d
+        self.logger.info(
```
Comment (lukas-bednar) [inline rrmngmnt/package_manager.py]: I think it doesn't deserve to be info level

File: rrmngmnt/package_manager.py [hunk 3]
```diff
@@ -81,34 +107,59 @@ def install(self, package):
                 "Install package %s on host %s", package, self.host
             )
             return self._run_command_on_host(cmd)
-        self.logger.info(
+        self. logger.info(
```
Comment (lukas-bednar) [inline rrmngmnt/package_manager.py]: Typo?

File: rrmngmnt/package_manager.py [hunk 4]
```diff
@@ -81,34 +107,59 @@ def install(self, package):
...
         :type package: str
-        :return: True, if package removal success, otherwise False
+        :param pattern: If True package name is pattern
+        :return: True, if package(s) removal success, otherwise False
         :rtype: bool
         :raise: NotImplementedError
         """
         if not self.remove_command_d:
-            raise NotImplementedError("There is no 'remove' command defined.")
+            raise NotImplementedError("there is no 'remove' command defined.")
```
Comment (lukas-bednar) [inline rrmngmnt/package_manager.py]: Why did you lowercase the fist letter? Sentence should start with capital letter ...

File: rrmngmnt/package_manager.py [hunk 5]
```diff
@@ -81,34 +107,59 @@ def install(self, package):
...
+            for package in package_list:
+                if not self._run_command_on_host(cmd + [package]):
+                    return False
+            return True
+
+        package_exists = self.exist(package)
+        if not package_exists:
+            self.logger.info(
+                "Package %s not exist on host %s", package, self.host
+            )
```
Comment (lukas-bednar) [inline rrmngmnt/package_manager.py]: ... does not ...

**Comment by lukas-bednar** [comment]
In general I like the list function, but I don't like that you use it for removing operation when pattern==true ... it will slow down that operation, and increase traffic which goes in wain since you thow it away at the end of operation.

**Comment by petr-balogh** [comment]
Unfortunately I pulled master instead of right branch i have locally and this PR is closed now.. should I open the new one or can you reopen it somehow?

**Pull request closed without merging**